src/actions/albumsActions.ts

```diff
@@ -267,14 +334,15 @@ export function fetchPlaceAlbumsList() {
   };
 }
 
-export function fetchPlaceAlbum(album_id) {
-  return function (dispatch) {
+export function fetchPlaceAlbum(album_id: string) {
+  return function (dispatch: Dispatch<any>) {
     dispatch({ type: "FETCH_PLACE_ALBUMS" });
     Server.get(`albums/place/${album_id}/`)
       .then((response) => {
+        const data = PlaceAlbumInfoSchema.parse(response.data)
```

Comment:
We have to add a .results to response.data.

Reply:
We then get the following zod error:

```
{
  "issues": [
    {
      "code": "invalid_type",
      "expected": "number",
      "received": "string",
      "path": [
        "id"
      ],
      "message": "Expected number, received string"
    },
    {
      "code": "invalid_type",
      "expected": "array",
      "received": "undefined",
      "path": [
        "cover_photos"
      ],
      "message": "Required"
    },
    {
      "code": "invalid_type",
      "expected": "number",
      "received": "undefined",
      "path": [
        "photo_count"
      ],
      "message": "Required"
    },
    {
      "code": "invalid_type",
      "expected": "number",
      "received": "undefined",
      "path": [
        "geolocation_level"
      ],
      "message": "Required"
    }
  ],
  "name": "ZodError"
}
```

Here is an examplary response:

```
"results": {
		"id": "14",
		"title": "Neukölln",
		"grouped_photos": [
			{
				"date": "2021-02-14",
				"location": "",
				"items": [
					{
						"id": "41e2399ff4ed78d5dc4bcbfabdd8b26f1",
						"dominantColor": "#6f829e",
						"url": "41e2399ff4ed78d5dc4bcbfabdd8b26f1",
						"location": "Wildenbruchplatz 12045 Neukölln Berlin Germany",
						"date": "2021-02-14T16:18:03+00:00",
						"birthTime": "2021-02-14 16:18:03+00:00",
						"aspectRatio": 1.33,
						"type": "image",
						"rating": 0,
						"owner": {
							"id": 1,
							"username": "admin",
							"first_name": "",
							"last_name": ""
						},
						"shared_to": []
					}
				]
			}
		]
	}
```

Should we fix the response from the backend to be more in line with the other data types?

Reply:
My intention was to just type them as they are, and that we can use these types to identify where we want to adjust the responses in the backend to have fewer different response types.

Reply:
I wonder how I could get this type so wrong...

Reply:
Ah, there is a serializer_class defined, but it's not used...

Reply:
Fixed.

Reply:
Yes, now it works 🥳